src/ports/postgres/modules/utilities/path.py_in

```diff
@@ -118,140 +120,175 @@ def path(schema_madlib, source_table, output_table, partition_expr,
-                                            FROM
-                                                {matched_partitions}
-                                            ) ssubq1
-                                        ) subq1
-                                    GROUP BY {p_col_name_str}, matched_ids
-                                ) q1
+                            FROM
+                                {length_of_between_matches} AS q1
                                 JOIN
-                                (
-                                    -- lengths of all matches
-                                    SELECT
-                                        {p_col_name_str},
-                                        matched_ids,
-                                        array_agg(length(matches) ORDER BY match_index) AS matches
-                                    FROM (
-                                        SELECT
-                                            {p_col_name_str},
-                                            matched_ids,
-                                            generate_series(1, num_matches) AS match_index,
-                                            (regexp_matches(
-                                                   sym_str, '(?i)({pattern_expr})', 'g'))[1] AS matches
-                                        FROM (
-                                            SELECT
-                                                {p_col_name_str},
-                                                matched_ids,
-                                                sym_str,
-                                                count(matches) AS num_matches
-                                            FROM (
-                                                SELECT
-                                                    {p_col_name_str},
-                                                    matched_ids,
-                                                    sym_str,
-                                                    (regexp_matches(
-                                                           sym_str, '(?i)({pattern_expr})', 'g'))[1] AS matches
-                                                FROM
-                                                   {matched_partitions}
-                                                ) t1
-                                            GROUP BY {p_col_name_str}, matched_ids, sym_str
-                                            ) t2
-                                        ) subq2
-                                    GROUP BY {p_col_name_str}, matched_ids
-                                ) q2
+                                {length_of_matches} as q2
                                 USING ({p_col_name_str})
-                            GROUP BY {p_col_name_str}, q1.matched_ids, match_splits, matches
+                            GROUP BY {p_col_name_str}, q1.{match_to_row_id}, match_splits, matches, match_indices
```

Comment:
actually the `group by` is an artifact of the old query. After the refactor, we don't need grouping here. good catch.